### DISCUSSION TITLE
No connection to Collabora Online - Built-in CODE Server

### DISCUSSION THREAD

**Discussion by N8schatten** [nextcloud/all-in-one#1119] [Questions] [answered]
Hi,
I got an new installation of Nextcloud AIO in a Docker Container on a Synology NAS (x64). When i install the Collabora Online - Built-in CODE Server and want to open a Document, the page is only loading and nothing else happens. In the nextcloud settings for Nextcloud Office it shows no connection to Collabora Online - Built-in CODE Server.
In an other installation with the legacy docker container from Nextcloud (not all-in-one) the Built-in CODE Server works fine.
(Reverse proxy an let's encrypt certificate for both containers managed form Hostsystem, Synology DSM 7.1-42661 Update 4)

Can someone please help me I can't get any further?
Attached Files: nextcloud log, docker compose file, and 2 pictures.

[docker-compose.txt](https://github.com/nextcloud/all-in-one/files/9490390/docker-compose.txt)
[nextcloud.log](https://github.com/nextcloud/all-in-one/files/9490391/nextcloud.log)
![nextcloud_collabora](https://user-images.githubusercontent.com/89748315/188467788-aa258d42-ab0b-4ea5-b823-61f12c077dc1.png)
![nextcloud_collabora_loading](https://user-images.githubusercontent.com/89748315/188467791-1c9643b6-20a2-4002-9f03-eb3bb0ea8f40.png)

**Comment by szaimen** [comment]
Hi, are you running AIO behind an Apache reverse proxy?

**Reply by N8schatten** [reply]
Hi, its behind the  reverse proxy form Synology, from the Hostsystem. It had to be nginx. with GUI.

I redirect there once the http (port 80)url to https port (443). And with a second rule the https url on "localhost port 8084.
(Other port, see docker compose attachment)

**Reply by N8schatten** [reply]
Tried to go directly from the url to localhost without the first rule (no http 80 to https 443). But without changing.

**Comment by szaimen** [comment]
Are websocket connections enabled?

**Reply by N8schatten** [reply]
![reverseproxy](https://user-images.githubusercontent.com/89748315/188865846-bf0dd93c-fbec-47cf-972c-aaadaf844a95.png)
Yes, they are

**Comment by N8schatten** [comment]
I also noticed that whenever I restart the containers, the app collabora online (richdocumens) is no longer installed.
I can then reinstall the app, but still the same error. After container restarts, the app is gone again.

**Comment by szaimen** [comment]
I see so simply try the Nextcloud Office option from the AIO interface? that should automatically enable configurre and install things.

**Comment by N8schatten** [comment]
Even if I select in AIO Collabora (Nextcloud Office) it does not work either with "use your own server" nor with buil in code server.
In addition, the complete nextcloud will be extremely slow

**Comment by szaimen** [comment]
> Even if I select in AIO Collabora (Nextcloud Office) it does not work either with "use your own server"

I see. Though is the connection green in that case?

**Comment by N8schatten** [comment]
![nextcloudoffice2](https://user-images.githubusercontent.com/89748315/189217460-49e30359-964c-4d1a-a889-82312378baa3.png)
![nextcloudoffice](https://user-images.githubusercontent.com/89748315/189217463-e2ea95f7-4b10-487b-9be6-d967ce472bbf.png)

**Comment by N8schatten** [comment]
When i reload the page th connection for build in code server is green, but when i push on it it gets red.

**Comment by szaimen** [comment]
Did you modify the url manually? It should set it automatically when the collabora option is chosen in the aio interface... Is the connection green with that?

**Comment by N8schatten** [comment]
no is unchanged. Do I have to redirect another port in my reverse proxy for collabora? except 443 to the apache port

**Comment by szaimen** [comment]
Ca you remove the richdcoumentscode app and restart the containers? It should set the url then automatically...

**Comment by N8schatten** [comment]
Or is it a problem that I use other ports in my docker compose file:

version: "3.8"

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer # This line is not allowed to be changed

services:
  nextcloud:
    image: nextcloud/all-in-one:latest # Must be changed to 'nextcloud/all-in-one:latest-arm64' when used with an arm64 CPU
    restart: always
    container_name: nextcloud-aio-mastercontainer # This line is not allowed to be changed
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config # This line is not allowed to be changed
      - /var/run/docker.sock:/var/run/docker.sock:ro # May be changed on macOS, Windows or docker rootless. See the applicable documentation
    ports:
      - 8083:8080 # Mastercontainer Interface with self-signed certificate 
    environment: # Is needed when using any of the options below
      - APACHE_PORT=8084 # Is needed when running behind a reverse proxy. See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
      - APACHE_IP_BINDING=127.0.0.1 # Should be set when running behind a reverse proxy that is running on the same host. See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
      - TALK_PORT=3478

**Comment by szaimen** [comment]
> Or is it a problem that I use other ports in my docker compose file:

No, that shouldn't be a problem.

**Comment by szaimen** [comment]
Please try this:

> Ca you remove the richdcoumentscode app and restart the containers? It should set the url then automatically...

**Comment by N8schatten** [comment]
The url is now simply the address of my nextcloud. But still no connection.

**Comment by szaimen** [comment]
> The url is now simply the address of my nextcloud.

Yes, that is how it should be

>  But still no connection.

Then is probably something wrong with your reverse proxy configuration

**Reply by N8schatten** [reply]
I redirect my url port 443 to localhost 8084 (apache port).

**Reply by szaimen** [reply]
> I redirect my url port 443 to localhost 8084 (apache port).

That is correct. I mean one of the other settings

**Comment by szaimen** [comment]
Or you are running into this issue: https://github.com/nextcloud/all-in-one/issues/834

However I didn't find a solution to it yet (and it doesnt look like it is an apache based reverse proxy)...

**Reply by N8schatten** [reply]
no, nginx

**Reply by szaimen** [reply]
Yes i thought so

**Comment by N8schatten** [comment]
But I would have preferred to use the build in code server but also this one is strangely not

**Reply by szaimen** [reply]
> But I would have preferred to use the build in code server but also this one is strangely not

Yes, this is an additional indication that something is incorrectly configured in your reverse proxy

**Comment by N8schatten** [comment]
Ok, then I will go a little further at this point. Will continue tomorrow. Thank you.

**Comment by N8schatten** [comment]
In a synology forum I found out that the standard collabora docker container probably does not run because no seccomp is available on the diskstation.
In addition, the nextcloud-aio-collabora container appears as unhelthy (in Portainer). Could that be the problem?

**Reply by szaimen** [reply]
> In a synology forum I found out that the standard collabora docker container probably does not run because no seccomp is available on the diskstation.

No idea about this.

> In addition, the nextcloud-aio-collabora container appears as unhelthy (in Portainer). Could that be the problem?

Most probably yes.

**Reply by N8schatten** [reply]
Ok, so you got no idea how to disable seccomp for nextcloud-aio-collabora container?

**Reply by szaimen** [reply]
> Ok, so you got no idea how to disable seccomp for nextcloud-aio-collabora container?

correct

**Comment by N8schatten** [comment]
So it's probably because the collabora container does not start properly.
In the log of the collabora container are several errors and among others:

cp: cannot create regular file '/opt/cool/systemplate/etc/resolv.conf': Permission denied
cp: cannot create regular file '/opt/cool/systemplate/etc/hosts': Permission denied

Maybe this is a problem with synology's docker integration... No idea.

**Reply by szaimen** [reply]
> Maybe this is a problem with synology's docker integration...

Yeah, sounds like it :/

**Comment by N8schatten** [comment]
Only a pity that the build in code server (from nextcloud appstore) does not work either. With the "normal" Docker image of nextcloud this works.

**Reply by szaimen** [reply]
I googled this and it seems like this may be because the Nextcloud container is Alpine Linix based and not Debain based so a ristriction of richdocumentscode...

**Reply by N8schatten** [reply]
Ok, thanks for the effort.
Then this can probably not be changed. What a pity

**Reply by szaimen** [reply]
You may try OnlyOffice instead...

<details>
<summary>Click here</summary>

See https://github.com/nextcloud/all-in-one/pull/1024/files?file-filters%5B%5D=.twig&show-viewed-files=true

</details>

**Reply by N8schatten** [reply]
Had tried. That works.

**Reply by N8schatten** [reply]
Will then probably stay with my old nextcloud setup until there is something new.

**Comment by N8schatten** [comment]
Solved the problem now.
It's because the collabora container requires "Seccomp". But Synology doesn't offer that.
[https://en.wikipedia.org/wiki/Seccomp](url)
If someone had the same problem. I did the following:

1.  Initialize (start) Nextclud and collabora normally via AIO interface
2.  Then stop the collabora-aio container **from the Synology docker app**
3.  Enter in the container settings of the collabora-aio container at extra_params --o:security.seccomp=false to the already existing 
    parameters.
4.  Start the container again **via synologie docker app**
Collaora will then run

Unfortunately, this has to be done again every time you restart from the AIO interface, as the settings will probably be overwritten.
So also with every update.

It would be nice if it were possible to keep this setting permanently in the future
Or maybe as a selection in the AIO interface, before the container is deployed.
.

**Comment by szaimen** [comment, marked as answer]
Allowing to disable seccomp will be handled in https://github.com/nextcloud/all-in-one/issues/1149

**Comment by szaimen** [comment]
@N8schatten btw, would you be able to create some screenshots for me that I can use as an example in our reverse proxy documentation for the Synology Reverse Proxy? You can look at nginx proxy manager in the documentatino which example values it could have...

**Reply by N8schatten** [reply]
Sure, I can do it. But is everything on german. 
Will do some screeshots tomorrow.

**Reply by szaimen** [reply]
Thank you! Would it be posaible to switcht the interface temporarily to english for the screenshots? If not, german acreenshots should do it as well. 👍

**Reply by N8schatten** [reply]
So here are some photos of the reverse proxy settings.

With the headers I'm not sure which ones are really needed so I have a picture only with websocket and one as I have it. Both work.
The ssl certificate is made via "Control Panel" > "Security" > "Certificate".

If anything else is needed or isn't right, let me know.

![reverseproxy_main](https://user-images.githubusercontent.com/89748315/192525606-48cab54b-866e-4964-90a8-15e71bd362fb.png)
![reverseproxy_header_wbsock](https://user-images.githubusercontent.com/89748315/192525635-a0aecae0-8fb5-436d-b9e2-d473379d298f.png)
![reverseproxy_header_all](https://user-images.githubusercontent.com/89748315/192525681-c06f3b39-f510-458e-b1f2-6b2cd995e24c.png)

**Reply by N8schatten** [reply]
Just tested again with only the websocket header. Everything works. So I would just leave those headers.(Upgrade, Connection)

**Reply by szaimen** [reply]
Thank you very much! I just added the provided files to the documentation: https://github.com/nextcloud/all-in-one/pull/1206/files?short_path=13df81b#diff-13df81bd3ba03aae4c206ef5abd710d2e6045e7cbf57fc6682c9d2ae08d17c2c

Does this look good to you? :)

**Reply by N8schatten** [reply]
But only the header for websocket is enough, like picture 2. So picture 2 would be better than picture 3. I Think.

**Reply by szaimen** [reply]
Not sure about this. Cannot test this myself so better safe than sorry.
